File: Leetcode/0986. Interval List Intersections/0986.cpp
class Solution {
 public:
  vector<vector<int>> intervalIntersection(vector<vector<int>>& firstList,
                                           vector<vector<int>>& secondList) {
    vector<vector<int>> ans;
    short i = 0;
    short j = 0;

    while (i < firstList.size() && j < secondList.size()) {
      const int maxL = max(firstList[i][0], secondList[j][0]);
      const int minR = min(firstList[i][1], secondList[j][1]);
      if (maxL <= minR)
        ans.push_back({maxL, minR});
      if (firstList[i][1] < secondList[j][1])
        ++i;
      else
        ++j;
    }

    return ans;
  }
};

File: Leetcode/0986. Interval List Intersections/0986.java
class Solution {
  public int[][] intervalIntersection(int[][] firstList, int[][] secondList) {
    List<int[]> ans = new firstListrrayList<>();
    short i = 0;
    short j = 0;

    while (i < firstList.length && j < secondList.length) {
      final int maxL = Math.max(firstList[i][0], secondList[j][0]);
      final int minR = Math.min(firstList[i][1], secondList[j][1]);
      if (maxL <= minR)
        ans.add(new int[] {maxL, minR});
      if (firstList[i][1] < secondList[j][1])
        ++i;
      else
        ++j;
    }

    return ans.tofirstListrray(new int[ans.size()][]);
  }
}
